### DISCUSSION TITLE
When I try to install with the composer I get this error

### DISCUSSION THREAD

**Discussion by adriano461997** [ctwillie/expo-server-sdk-php#2] [Q&A] [answered]
When I try to install with composer running` composer require ctwillie/expo-server-sdk-php` I get this error

```
  Problem 1
    - Root composer.json requires ctwillie/expo-server-sdk-php ^1.1 -> satisfiable by ctwillie/expo-server-sdk-php[v1.1].
    - ctwillie/expo-server-sdk-php v1.1 requires guzzlehttp/guzzle ^7.3 -> found guzzlehttp/guzzle[7.3.0] but the package is fixed to 6.5.5 (lock file version) by a partial update and that version does not match. Make sure you list
it as an argument for the update command.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

```
what to do?

**Comment by Razorholt** [comment]
I just managed to install it with this "composer require ctwillie/expo-server-sdk-php -W"
However, when I tried the script I got this error: 

`Fatal error: Uncaught Error: Class 'Expo' not found in /home/project/public_html/testpush.php:5 Stack trace: #0 {main} thrown in /home/project/public_html/testpush.php on line 5`

I did load autoload.php BTW

**Reply by ctwillie** [reply]
@Razorholt If you are sure your have configured autoloading correctly, did you require the class in ```testpush.php```?

```
// in testpush.php

require __DIR__ . '/vendor/autoload.php';

use ExpoSDK\Expo;

**Reply by Razorholt** [reply]
Yes, I did that

**Comment by ctwillie** [comment, marked as answer]
@adriano461997 
It looks like you already had ```guzzlehttp/guzzle``` installed in your project and there is a version conflict. I would see if you could get your project to only use guzzle version ```^7.3``` instead of having the version fixed to ```6.5.5```. Unless there is a good reason to only use that specific version.

**Reply by Razorholt** [reply]
Got it to work. Thanks a lot, man!

**Comment by Razorholt** [comment]
Why am I getting this when I add `$data = $response->getData();` ?

`Fatal error: Uncaught Error: Call to undefined method stdClass::getData() in /home/project/public_html/testpush2.php:35 Stack trace: #0 {main} thrown in /home/project/public_html/testpush2.php on line 35`

getReceipts() also gives me the stdClass error. Here is my code;


```
require '/home/project/vendor/autoload.php';

use ExpoSDK\ExpoMessage;
use ExpoSDK\Expo;

$expo = new Expo();

// composed message, see above
$message = (new ExpoMessage())
    ->setTitle('Message Title')
    ->setSubtitle('Message sub title')
    ->setBody('The notification message body')
    ->setChannelId('default')
	->setData(['foo' => 'bar'])
    ->setBadge(0)
    ->playSound();

$recipients = [
    'ExponentPushToken[********************]'
];

// send
$response = $expo->send($message)->to($recipients)->push();
```

**Reply by ctwillie** [reply]
@Razorholt Try running ```composer update ctwillie/expo-server-sdk-php``` to pull in the newest release that handles responses. Let me know if that resolves this issue.

**Reply by Razorholt** [reply]
Works like a charm! Thank you so much!

EDIT: I will test the channels next and will report back if I find any bugs.

**Comment by Razorholt** [comment]
If I may ask another question.... 

`$res = $expo->getReceipts($ticketIds);` gives me that array:

```
ExpoSDK\ExpoResponse Object
(
    [response:ExpoSDK\ExpoResponse:private] => Array
        (
            [data] => Array
                (
                )
        )
    [statusCode:ExpoSDK\ExpoResponse:private] => 200
)
```

1.- How do I get the statusCode response only? I tried to json_emcode and do something like that: $res["statusCode"] but it doesn't work
2.- Same for data. And what kind of results should I get within that data array? A text or code explaining why the message hasn't been delived?

Thanks, man!

**Reply by ctwillie** [reply]
@Razorholt 

1. To get the response status code call ```$response->getStatusCode()```.
2. The structure of the response data can vary. Too much to explain here, but I would recommend reading the Expo docs [found here](https://docs.expo.dev/push-notifications/sending-notifications/) to understand what may be returned from their servers.

**Reply by Razorholt** [reply]
Excellent! Thank you so much!

**Comment by Razorholt** [comment]
why is getStatusCode() giving you a 200 response no matter what? 

```
$ticketIds = ['whatever'];
$res = $expo->getReceipts($ticketIds);
$res->getStatusCode(); // 200
```

```
// print_r($res)

ExpoSDK\ExpoResponse Object
(
    [response:ExpoSDK\ExpoResponse:private] => Array
        (
            [data] => Array
                (
                )
        )
    [statusCode:ExpoSDK\ExpoResponse:private] => 200
)
```

**Reply by ctwillie** [reply]
@Razorholt ```getStatusCode()``` simply returns the status code returned by the Expo servers. The data and status codes they return from their servers is explained in the Expo docs.
